lib/main.js

```diff
@@ -39,21 +38,18 @@ export default {
       lintOnFly: true,
       lint: (activeEditor) => {
         const command = atom.config.get("linter-ruby.rubyExecutablePath");
-        const file = activeEditor.getPath();
-        const cwd = path.dirname(file);
-        const args = ["-wc", file];
-        return helpers.exec(command, args, {stream: "stderr", cwd: cwd}).then(output => {
-          const toReturn = [];
+        return helpers.exec(command, ['-wc'], {stdin: activeEditor.getText(), stream: 'stderr'}).then(output => {
+          var toReturn = [];
           output.split(/\r?\n/).forEach(function (line) {
             const matches = regex.exec(line);
             if (matches === null) {
               return;
             }
             toReturn.push({
-              range: helpers.rangeFromLineNumber(activeEditor, Number.parseInt(matches[1])),
-              type: "Error",
+              range: helpers.rangeFromLineNumber(activeEditor, Number.parseInt((matches[1] - 1))),
+              type: 'Error',
               text: matches[2],
-              filePath: file
+              filePath: activeEditor.getPath()
```

Comment:
can you store this in a variable instead of calling a function for each message?